From 103c76a06545a4dcf62b6985a6fc82b298b29ce9 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Mon, 13 Nov 2023 10:46:54 +0100
Subject: [PATCH 0621/1695] plat-stm32mp2: conf: enable single core for
 STM32MP21

Force CFG_TEE_CORE_NB_CORE to 1 for STM32MP21 series.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 834ff27fdf2af6e7057d6511826b0fcfb93d65d7 plat-stm32mp2: conf: enable single core for STM32MP21]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I5325d750c136a996b700122ac11190d94258d425
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361572
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index ad5162762..bf58536b5 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -109,6 +109,9 @@ CFG_DTB_MAX_SIZE ?= 262144
 CFG_HALT_CORES_ON_PANIC ?= y
 CFG_MMAP_REGIONS ?= 30
 CFG_NUM_THREADS ?= 5
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP21)),)
+$(call force,CFG_TEE_CORE_NB_CORE,1)
+endif
 CFG_TEE_CORE_NB_CORE ?= 2
 CFG_STM32MP_OPP_COUNT ?= 3
 
-- 
2.47.2

